DVSRegister/Middleware/SecurityHeadersMiddleware.cs

```diff
@@ -2,29 +2,50 @@
 using System.Security.Policy;
 using System.Text;
 using DVSRegister.CommonUtility;
+using Microsoft.AspNetCore.Html;
 using Microsoft.Extensions.Options;
 
 namespace DVSRegister.Middleware
 {
     public class SecurityHeadersMiddleware
     {
         private readonly RequestDelegate _next;
+        public const string NonceContextKey = "CSP-Nonce";
         public SecurityHeadersMiddleware(RequestDelegate next)
         {
             _next = next;//Storing the reference to next middleware in pipeline
         }
 
         public async Task InvokeAsync(HttpContext context)
         {
+            // Generate a nonce for the current request
+            var nonce = Guid.NewGuid().ToString("N");
+            context.Items[NonceContextKey] = nonce;
+
             if (!context.Response.HasStarted)
             {
                 // Added security headers
                 context.Response.Headers["X-Frame-Options"] = "DENY";
-                context.Response.Headers["Content-Security-Policy"] = "frame-ancestors 'none'";
-            }
 
+                //CSP with nonce for inline scripts
+                context.Response.Headers["Content-Security-Policy"] =
+                $"script-src 'nonce-{nonce}' 'unsafe-inline' 'self' https:; " +
```

Comment:
when we add nonce need to call the method in inline scripts or styles that are used in our project
We need to search for   <script > or <style > inline tags in cshtml pages and call GetScriptNoncemethod